### ISSUE TITLE
Deserialisation via pickle fails for trained model

### ISSUE THREAD

**Issue by alexallan** [catboost/catboost#173]
After training a binary classification problem, the saved model cannot be deserialised once serialised. This behaviour is not reproduced if the model is not trained. I am using catboost 0.4.1

```
>>> pkl.loads(pkl.dumps(model))
Traceback (most recent call last):
  File "/usr/local/lib/python3.6/site-packages/_pydevd_bundle/pydevd_exec2.py", line 3, in Exec
    exec(exp, global_vars, local_vars)
  File "<input>", line 1, in <module>
  File "_catboost.pyx", line 812, in _catboost._CatBoostBase.__setstate__ (/home/donskov/.ya/build/build_root/028i/000365/catboost/python-package/catboost/_catboost.pyx.cpp:14836)
  File "_catboost.pyx", line 844, in _catboost._CatBoostBase._set_test_eval (/home/donskov/.ya/build/build_root/028i/000365/catboost/python-package/catboost/_catboost.pyx.cpp:15850)
  File "_catboost.pyx", line 679, in _catboost._CatBoost._set_test_eval (/home/donskov/.ya/build/build_root/028i/000365/catboost/python-package/catboost/_catboost.pyx.cpp:10841)
  File "_catboost.pyx", line 682, in _catboost._CatBoost._set_test_eval (/home/donskov/.ya/build/build_root/028i/000365/catboost/python-package/catboost/_catboost.pyx.cpp:10694)
TypeError: 'float' object is not iterable
```

**Comment by annaveronika** [comment]
We changed to flat buffers in 0.4, so if it's trained with version before that it will not deserialize with a new version.

What does it mean "not reproduced if the model is not trained?"
Could you provide the code you are running?

**Comment by annaveronika** [comment]
closing because of inactivity

**Comment by mmortazavi** [comment]
Here facing the same problem with Catboost 0.5.2.1. Basically CatBoostRegressor is built, and saved (serialized) to have persistent model via either scikit learn pickle of joblid. But during loading the model (deserialization) it returns with the error:

TypeError: 'float' object is not iterable

**Comment by pashna** [comment]
Have the same issue as @mmortazavi mentioned with version 0.6.1.1

**Comment by pashna** [comment]
Would appreciate any ideas for a quick hacky fix :)

**Comment by stasSajin** [comment]
Seeing this issue in `0.8.1 `. It would be really great to have a fix for this issue since it is difficult to apply any native python serialization protocols. 

I've tried both dill and pickle implementations using a variety of protocol and they all led to the same Typeerror.

**Comment by kizill** [comment]
Hi everyone, we fixed this while adding multiple eval set support. The fix available in master and will be included in 0.9 release.
